src/main/java/com/gotocompany/depot/redis/RedisSink.java

```diff
@@ -35,15 +36,35 @@ public SinkResponse pushToSink(List<Message> messages) {
         List<RedisRecord> validRecords = splitterRecords.get(Boolean.TRUE);
         SinkResponse sinkResponse = new SinkResponse();
         invalidRecords.forEach(invalidRecord -> sinkResponse.addErrors(invalidRecord.getIndex(), invalidRecord.getErrorInfo()));
-        if (validRecords.size() > 0) {
-            List<RedisResponse> responses = redisClient.send(validRecords);
-            Map<Long, ErrorInfo> errorInfoMap = RedisSinkUtils.getErrorsFromResponse(validRecords, responses, instrumentation);
+        if (!validRecords.isEmpty()) {
+            Map<Long, ErrorInfo> errorInfoMap = send(validRecords);
             errorInfoMap.forEach(sinkResponse::addErrors);
             instrumentation.logInfo("Pushed a batch of {} records to Redis", validRecords.size());
         }
         return sinkResponse;
     }
 
+    private Map<Long, ErrorInfo> send(List<RedisRecord> validRecords) {
+        List<RedisResponse> responses = null;
+        RuntimeException exception = null;
+        int retry = CONNECTION_RETRY;
+        while (retry > 0) {
+            try {
+                responses = redisClient.send(validRecords);
+                break;
+            } catch (RuntimeException e) {
+                exception = e;
```

Comment:
let's log it here. Because we might not see any exception if the next retry is successful.

Reply:
fixed